tools/droplets/fabric/fabfile.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import fabric
+from fabric.api import run, put, puts, task, env
+import os
+
+env.user = "root"
+
+def clean_up() -> None:
+    """
+    Clean up remote machine before taking snapshot.
+    """
+    run("rm -rf /tmp/* /var/tmp/*")
+    run("history -c")
+    run("cat /dev/null > /root/.bash_history")
+    run("unset HISTFILE")
+    run("apt-get -y autoremove")
+    run("apt-get -y autoclean")
+    run(br"find /var/log -mtime -1 -type f -exec truncate -s 0 {} \;")
+    run("rm -rf /var/log/*.gz /var/log/*.[0-9] /var/log/*-????????")
+    run("rm -rf /var/lib/cloud/instances/*")
+    run("rm -rf /var/lib/cloud/instance")
+    puts("Removing keys...")
+    run("rm -f /root/.ssh/authorized_keys /etc/ssh/*key*")
+    run("dd if=/dev/zero of=/zerofile; sync; rm /zerofile; sync")
+    run("cat /dev/null > /var/log/lastlog; cat /dev/null > /var/log/wtmp")
+
+def install_files() -> None:
+    """
+    Install files onto remote machine.
+    Walk through the files in the "files" directory and copy them to the build system.
+    File permissions will be inherited.  If you need to change permissions on uploaded files
+    you can do so in a script placed in the "scripts" directory.
+    """
+    print("--------------------------------------------------")
+    print("Copying files in ./files to remote server")
+    print("--------------------------------------------------")
+    rootDir = './files'
+    for dirName, subdirList, fileList in os.walk(rootDir):
+        cDir = dirName.replace("./files", "")
+        print("Entering Directory: {}".format(cDir))
+        if cDir:
+            run("mkdir -p {}".format(cDir))
+        for fname in fileList:
+            cwd = os.getcwd()
+            rpath = cDir + "/" + fname
+            lpath = cwd + "/files" + cDir + "/" + fname
+            print('Moving File: {}'.format(lpath))
+            put(lpath, rpath, mirror_local_mode=True)
+
+def run_scripts() -> None:
+    """
+    Run all scripts in the "scripts" directory on the build system
+    Scripts are run in alpha-numeric order.  We recommend naming your scripts
+    with a name that starts with a two digit number 01-99 to ensure run order.
+    """
+    print("--------------------------------------------------")
+    print("Running scripts in ./scripts")
+    print("--------------------------------------------------")
+
+    cwd = os.getcwd()
+    directory = cwd + "/scripts"
+
+    for f in os.listdir(directory):
+        lfile = cwd + "/scripts/" + f
+        rfile = "/tmp/" + f
+        print("Processing script in {}".format(lfile))
+        put(lfile, rfile)
+        run("chmod +x {}".format(rfile))
+        run(rfile)
+
+def setup_interactive_login_script() -> None:
```

Comment:
Added this function

Reply:
So I think a good way to do this is just to have 2 commits: First one that adds their code verbatim, and a second that makes our changes.  That'd be a lot more readable.

Reply:
(And much easier to rebase on top of a future version of their library)